src/operator/src/statement/ddl.rs

```diff
@@ -907,6 +908,21 @@ impl StatementExecutor {
                     violated: format!("Invalid table name: {}", new_table_name)
                 }
             );
+        } else if let AlterKind::AddColumns { columns } = alter_kind {
+            // If all the columns are marked as add_if_not_exists and they already exist in the table,
+            // there is no need to perform the alter.
+            let column_names: HashSet<String> = table_info
+                .meta
+                .schema
+                .column_schemas()
+                .iter()
+                .map(|schema| schema.name.clone())
```

Comment:
nitpick: Is possible to collect the `&String` or `&str` into the set?
```suggestion
            let column_names: HashSet<_> = table_info
                .meta
                .schema
                .column_schemas()
                .iter()
                .map(|schema| &schema.name)
```